tests/lib/rules/no-invalid-html-attribute.js

```diff
@@ -537,6 +560,73 @@ ruleTester.run('no-invalid-html-attribute', rule, {
         },
       ],
     },
+    {
+      code: '<link rel="shortcut"></link>',
+      errors: [
+        {
+          messageId: 'notAlone',
+          data: {
+            reportingValue: 'shortcut',
+            missingValue: 'icon',
+          },
+        },
+      ],
+    },
+    {
+      code: '<link rel="shortcut foo"></link>',
+      output: '<link rel="shortcut "></link>',
```

Comment:
```suggestion
      output: '<link rel="shortcut"></link>',
```

Reply:
Hmm but it actually does that - it's the opposite of

```js
      code: '<a rel={"batgo noopener"}></a>',
      output: '<a rel={" noopener"}></a>',
```

Reply:
```
 1) no-invalid-html-attribute
       invalid
         <link rel="shortcut foo"></link>:

      AssertionError [ERR_ASSERTION]: Output is incorrect.
      + expected - actual

      -<link rel="shortcut "></link>
      +<link rel="shortcut"></link>
```

The test fails when I remove it

Reply:
Might already be covered by

```js
      code: '<a rel="noreferrer noopener foobar"></a>',
      output: '<a rel="noreferrer noopener "></a>',
```
Same behaviour - I could just remove it maybe?

Reply:
Ahh no, we shouldn't remove it - it is the testcase for the `notPaired` branch (in contrast to `notAlone`)

Reply:
I'm basically saying that the autofix output should never have leading or trailing spaces.

Reply:
Yeah that also confused me, but apparently that's "how things used to be" in `master` as well:

https://github.com/yannickcr/eslint-plugin-react/blob/0d8946212a1ba560b12e2e01807033e19dc774f7/tests/lib/rules/no-invalid-html-attribute.js#L355

https://github.com/yannickcr/eslint-plugin-react/blob/0d8946212a1ba560b12e2e01807033e19dc774f7/tests/lib/rules/no-invalid-html-attribute.js#L378

https://github.com/yannickcr/eslint-plugin-react/blob/0d8946212a1ba560b12e2e01807033e19dc774f7/tests/lib/rules/no-invalid-html-attribute.js#L403

https://github.com/yannickcr/eslint-plugin-react/blob/0d8946212a1ba560b12e2e01807033e19dc774f7/tests/lib/rules/no-invalid-html-attribute.js#L416

https://github.com/yannickcr/eslint-plugin-react/blob/0d8946212a1ba560b12e2e01807033e19dc774f7/tests/lib/rules/no-invalid-html-attribute.js#L477

Apparently it only runs one cycle when fixing and we would need to find a way to kind of `while` loop so long until it can't fix any more issues - but that sounds like a major refactor to me 🤔

Reply:
fair point. altho it seems like adding `.trim()` to the attribute contents wouldn't be that difficult?

Reply:
Good idea! But where would I run this, I think I haven't fully grasped when/how fix runs. Would this be part of fix? Or would we just always in the end after all checks run another trim as it's never good/valid to keep it?

Reply:
Adding a single fixer on program exit that trimmed any attribute strings that had been previously autofixed (we’d have to track that manually) would be good. I think that should wait for a followup PR, tho, because it was a pre-existing problem. 

I’ll land this tomorrow, as-is.